### PR TITLE
Add enableAutoDois field to .dockstore.yml and allow disabling auto DOIs

### PR DESCRIPTION
**Pull request by kathy-t**
**Description**
This PR reuses code from https://github.com/dockstore/dockstore/pull/5996. It adds an `enableAutoDois` boolean that can only be set to `false` to the .dockstore.yml so that auto DOI generation can be disabled for GitHub App workflowish entries. If present, the workflow will have it's auto DOI generation disabled and it will remain disabled unless we use the admin endpoint to enable it again. When we remove the admin endpoint, we'll want to allow the enabling of auto DOI generation through this property if the user changes their mind and wants it enabled again.

**Review Instructions**
Requires a hotfix deploy. In staging:
- Enable auto DOI generation for your published GitHub App workflow using the admin endpoint.
- Create a tag and verify that a DOI was automatically created.
- Add `enableAutoDois: false` to your workflow.
- Create a tag and verify that a DOI was NOT created.
- Try setting `enableAutoDois: true`. Check the GitHub App logs and verify that the push failed because `enableAutoDois` can only be set to false.

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-6805

**Security and Privacy**

If there are any concerns that require extra attention from the security team, highlight them here and check the box when complete. 

- [x] Security and Privacy assessed

e.g. Does this change...
* Any user data we collect, or data location?
* Access control, authentication or authorization?
* Encryption features?

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [x] If you are changing dependencies, check the Snyk status check or the dashboard to ensure you are not introducing new high/critical vulnerabilities
- [x] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [x] Do not serve user-uploaded binary images through the Dockstore API
- [x] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [x] Do not create cookies, although this may change in the future
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java [hunk 1]
```diff
@@ -961,6 +961,11 @@ private WorkflowAndExisted createOrGetWorkflow(Class workflowType, String reposi
             }
         }
 
+        // Update the automatic DOI creation setting if it's been disabled. Enabling is currently controlled by an admin endpoint
+        if (Boolean.TRUE.equals(wf.getDisableDoiGeneration())) {
+            workflowToUpdate.setAutoGenerateDois(!wf.getDisableDoiGeneration());
```
Comment (denis-yuen) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java:966]: Seems `!wf.getDisableDoiGeneration()` is simpler as Boolean.FALSE here

File: dockstore-common/src/main/java/io/dockstore/common/yaml/AbstractYamlService.java [hunk 2]
```diff
@@ -158,6 +161,15 @@ public void setLatestTagAsDefault(boolean latestTagAsDefault) {
         this.latestTagAsDefault = latestTagAsDefault;
     }
 
+    @AssertTrue
```
Comment (denis-yuen) [inline dockstore-common/src/main/java/io/dockstore/common/yaml/AbstractYamlService.java:164]: Can you javadoc all these places that can be removed when we turn automatic DOI across the board with the same searchable comment, could be e.g.
`//  SEAB-6805]`

File: dockstore-common/src/main/java/io/dockstore/common/yaml/Workflowish.java [hunk 3]
```diff
@@ -17,6 +17,7 @@ public interface Workflowish {
     Filters getFilters();
     List<String> getTestParameterFiles();
     String getPrimaryDescriptorPath();
+    Boolean getDisableDoiGeneration();
```
Comment (denis-yuen) [inline dockstore-common/src/main/java/io/dockstore/common/yaml/Workflowish.java:20]: Should this be `boolean`, do we want to allow the possibility for null here?

**Review by denis-yuen — COMMENTED** [review]
minor question

File: dockstore-common/src/main/java/io/dockstore/common/yaml/Workflowish.java (hunk 3 above)
Comment (coverbeck) [reply dockstore-common/src/main/java/io/dockstore/common/yaml/Workflowish.java:20]: Boolean reflects what it is in the YAML, either a value of true or the property doesn't exist.  But we could make it a boolean here and have the implementations map the Boolean property to a boolean in the interface.

**Review by coverbeck — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]

**Review by svonworl — APPROVED** [review]
This is good, no need to change.  Some thoughts, though:
* we could simplify by just setting the database flag when set in `.dockstore.yml`, to the specified value. That way, it works both ways, no need for future mods, like magic!
* we could remove one logical inversion by naming the field something like "enableAutoDois"
* if the yaml field name contained the phrase "auto" or something similar, it would be clearer that you're enabling/disabling that particular type of DOI.

**Comment by svonworl** [comment]
Editorial: entry-wide properties in `.dockstore.yml` are confusing and users won't do it right, which is especially troublesome in a situation like disabling auto DOIs because you really don't want them to be generated.  Easy to miss an incorrect .dockstore.yml on a branch and tag it.

**Comment by kathy-t** [comment]
Changed the property name from `disableDoiGeneration` to `enableAutoDois` following PR feedback. Also created a documentation ticket https://ucsc-cgl.atlassian.net/browse/SEAB-6833

**Review by svonworl — APPROVED** [review]

**Review by coverbeck — APPROVED** [review]

**Pull request merged**